pkg/build/api/v1/types.go

```diff
@@ -138,6 +138,8 @@ const (
 	BuildSourceBinary BuildSourceType = "Binary"
 	// BuildSourceImage indicates the build will accept an image as input
 	BuildSourceImage BuildSourceType = "Image"
+	// BuildSourceNone indicates the build has no predefined input (only valid for Source and Custom Strategies)
+	BuildSourceNone BuildSourceType = "None"
```

Comment:
On Wed, Mar 23, 2016 at 4:21 PM, Clayton Coleman <notifications@github.com>
wrote:

> In pkg/build/api/v1/types.go
> <https://github.com/openshift/origin/pull/8219#discussion_r57229760>:
>
> > @@ -138,6 +138,8 @@ const (
> >  	BuildSourceBinary BuildSourceType = "Binary"
> >  	// BuildSourceImage indicates the build will accept an image as input
> >  	BuildSourceImage BuildSourceType = "Image"
> > +	// BuildSourceNone indicates the build has no predefined input (only valid for Source and Custom Strategies)
> > +	BuildSourceNone BuildSourceType = "None"
>
> Aren't we going to get rid of the BuildSourceType in the future? Is this
> to work around validations we already have in place?
>
Yes and Yes.


> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8219/files/7a520b79f9f513395be8894d0bc72360519a7ed6#r57229760>
>